homework/src/index.js

```diff
@@ -31,17 +33,116 @@
   }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { html: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { html: JSON.stringify(data, null, 2) });
-      }
+    const root = document.getElementById("root");
+
+    const div = createAndAppend("div", root);
+    const header = createAndAppend("p", div, {
+      text: "HYF Repositories",
+      class: "header"
     });
+
+    const table = createAndAppend("table", root, { class: "table" });
+    const select = createAndAppend("select", header, { class: "select" });
+
+    const tbody = createAndAppend("tbody", table);
+    const tr = createAndAppend("tr", tbody);
+    const container = createAndAppend("div", root, { class: "container" });
+    createAndAppend("td", tr, { text: "Repository: ", class: "label" });
+    const repositoryLink = createAndAppend("a", tbody);
+    const description = createAndAppend("tr", tbody);
+    const forks = createAndAppend("tr", tbody, { class: "label" });
+
+    const updated = createAndAppend("tr", tbody, { class: "label" });
+
+    fetchJSON(url)
+      .catch(error => {
+        createAndAppend("div", root, {
+          text: error.message,
+          class: "alert-error"
+        });
+        return;
+      })
+      .then(repositories => {
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+
+        repositories.forEach((element, index) => {
+          createAndAppend("option", select, {
+            text: element.name,
+            value: index
+          });
+        });
+
+        renderRepositoryBox(
+          repositories[0],
+          repositoryLink,
+          forks,
+          description,
+          updated
+        );
+
+        select.addEventListener("change", () => {
+          renderRepositoryBox(
+            repositories[select.value],
+            repositoryLink,
+            forks,
+            description,
+            updated
+          );
+        });
+        function renderRepositoryBox(
+          repositories,
+          repositoryLink,
+          forks,
+          description,
+          updated
+        ) {
+          repositoryLink.innerText = repositories.name;
+          repositoryLink.setAttribute("href", repositories.html_url);
+          repositoryLink.setAttribute("target", "_blank");
+
+          forks.innerHTML = "Forks: " + repositories.forks;
+          description.innerHTML = "Description: " + repositories.description;
```

Comment:
you sre using too much `innerHTML` and `setAttribute` while you can create your tags here and for each of them you can set the `text` as your change in `createAndAppend` function right?